src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/UkrlpData/UkrlpProfile.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+
+using AutoMapper;
+
+namespace SFA.DAS.RoatpCourseManagement.Application.UkrlpData
+{
+    public class UkrlpProviderDetailsProfile : Profile
+    {
+        public UkrlpProviderDetailsProfile()
+        {
+            CreateMap<MatchingProviderRecords, ProviderDetails>()
+                .ForMember(dest => dest.UKPRN,
+                    opt => opt.MapFrom(source => source.UnitedKingdomProviderReferenceNumber))
+                .ForMember(dest => dest.ProviderName, opt => opt.MapFrom(source => source.ProviderName))
+                .ForMember(dest => dest.ProviderStatus, opt => opt.MapFrom(source => source.ProviderStatus))
+                .ForMember(dest => dest.ContactDetails, opt => opt.MapFrom(source => source.ProviderContacts))
+                .ForMember(dest => dest.VerificationDate, opt => opt.MapFrom(source => source.ProviderVerificationDate))
+                .ForMember(dest => dest.ProviderAliases, opt => opt.MapFrom(source => source.ProviderAliases))
+                .ForMember(dest => dest.VerificationDetails, opt => opt.MapFrom(source => source.VerificationDetails));
+        }
+    }
+
+    public class UkrlpProviderContactProfile : Profile
+    {
+        public UkrlpProviderContactProfile()
+        {
+            CreateMap<ProviderContactStructure, ProviderContact>()
+                .ForMember(dest => dest.ContactType, opt => opt.MapFrom(source => source.ContactType))
+                .ForMember(dest => dest.ContactAddress, opt => opt.MapFrom(source => source.ContactAddress))
+                .ForMember(dest => dest.ContactPersonalDetails,
+                    opt => opt.MapFrom(source => source.ContactPersonalDetails))
+                .ForMember(dest => dest.ContactRole, opt => opt.MapFrom(source => source.ContactRole))
+                .ForMember(dest => dest.ContactTelephone1, opt => opt.MapFrom(source => source.ContactTelephone1))
+                .ForMember(dest => dest.ContactTelephone2, opt => opt.MapFrom(source => source.ContactTelephone2))
+                .ForMember(dest => dest.ContactWebsiteAddress,
+                    opt => opt.MapFrom(source => source.ContactWebsiteAddress))
+                .ForMember(dest => dest.ContactEmail, opt => opt.MapFrom(source => source.ContactEmail))
+                .ForMember(dest => dest.LastUpdated, opt => opt.MapFrom(source => source.LastUpdated));
+        }
+    }
+
+    public class UkrlpProviderAliasProfile : Profile
+    {
+        public UkrlpProviderAliasProfile()
+        {
+            CreateMap<ProviderAliasesStructure, ProviderAlias>()
+                .ForMember(dest => dest.Alias, opt => opt.MapFrom(source => source.ProviderAlias))
+                .ForMember(dest => dest.LastUpdated, opt => opt.MapFrom(source => source.LastUpdated));
+        }
+    }
+
+    public class UkrlpContactAddressProfile : Profile
+    {
+        public UkrlpContactAddressProfile()
+        {
+            CreateMap<ProviderContactAddress, ContactAddress>()
```

Comment:
We only need address, so I am confused why we need all other bits to be converted?

Reply:
I just lifted the code used in roatp-service and tried to keep our code as similar as possible, as I thought diverging would allow the likelihood of further code divergence.  However, I see what you mean, it feel like a lot of stuff we don't need, so I am now reworking both the mapping and simplifying to our core requirement to deliver the story.